## Simulations

Simulations were carried out to further examine the sensitivity of the JM estimations under several scenarios. 
Parameters were set based on values estimated from the strongest SNPs associated with T2D (Table \@ref(tab:SimulationSettings)), i.e. `r bestEffect[, "snp"]` in gene _`r bestEffect[, "closestgene"]`_ (T2D risk allele: `r bestEffect[, "effect_allele"]`; $OR=`r exp(bestEffect[, "effect.T2D"])`$; $\text{p-value}=`r formatScientific(bestEffect[, "pvalue.T2D"])`$ [@morris_large-scale_2012]; FPG increasing allele: `r bestEffect[, "effect_allele"]`; $\beta=`r bestEffect[, "effect.FG"]`$; $\text{p-value}=`r formatScientific(bestEffect[, "pvalue.FG"])`$ [@dupuis_new_2010]).

Longitudinal data were simulated according to Equation `r labEq.docx("eq:LongiFull")`, while event times were generated from an exponential distribution for the CoxPH model [@austin_generating_2012]
`r numberEq.docx("\\lambda_0(t)=\\lambda", "eq:Hazard")`
`r numberEq.docx("H_i(T_i)=\\int_0^{T_i}\\lambda \\exp(\\beta X_i(t)+\\alpha Z_i)dt", "eq:HazardDeriv")`
`r numberEq.docx("T_i=\\frac{1}{\\beta\\theta_{1i}}\\log\\left(1-\\frac{\\beta\\theta_{1i}\\times \\log(1-u)}{\\lambda \\exp(\\beta\\theta_{0i}+(\\beta\\gamma+\\alpha)Z_i)}\\right)", "eq:SurvTime")`

where $\lambda$ was set to achieve the targeted incidence rate in the simulated dataset.  
Datasets were simulated by varying the number of longitudinal measurements $m\in\{`r paste(ntimesvalues, collapse = "; ")`\}$, the number of individuals $n\in\{`r paste(format(nsamplevalues, big.mark = ","), collapse = "; ")`\}$, the allele frequency $f\in\{`r paste(mafvalues, collapse = "; ")`\}$ and the incidence rate $d\in\{`r paste(percincident, collapse = "; ")`\}$, thereby leading to `r length(nsamplevalues)*length(mafvalues)*length(ntimesvalues)*length(percincident)` different scenarios. 
Each scenario was simulated `r nKeep` times.

The Root-Mean Square Error (RMSE)
`r numberEq.docx("\\text{RMSE}(\\hat{\\theta})=\\sqrt{E((\\hat{\\theta}-\\theta)^2)}", "eq:SurvTime")`

was used to assess precision of estimators of $\beta$, $\gamma$ and $\alpha$, when testing the association between $Y_{ij}$, and $T_i$, the effect of Z_i$ on $Y_{ij}$ and the effect of $Z_i$ on $T_i$, respectively. 
In addition, statistical power and type I error were also estimated. 
The computational burden of each approach (JM and TS) was also investigated as our goal is to implement these approaches at a genome-wide scale.
