# Discussion

With the ever-increasing availability of genomic data generated by genotyping arrays and next generation sequencing, the need to develop and implement efficient models is important to ensure that statistical analysis will be achieved in a reasonable timeframe. 
In this paper, we proposed a comparison of two approaches, namely the joint model (JM) and the two-step model (TS), to estimate parameters accounting simultaneously for a genetic effect on both longitudinal and survival processes without discarding missing values or dropouts commonly generated by the longitudinal measurement process. 
In our real data application, FPG serves as the longitudinal process, whereas T2D diagnosis generates survival times of interest, both stochastic phenomenon being linked by the fact that a fixed threshold for FPG defines T2D onset (currently,  $[FPG]\geq 7$  mmol/L), along with glucose lowering medication use. 
Through simulations over different scenarios, we showed that joint models are less biased than classical separate approaches. Hopefully, joint models could provide more insight regarding the event of interest, and could assess the potential impact of a genetic marker on incident T2D better than simpler models.

By looking at statistical measures of accuracy such as RMSE for our model estimators, and by estimating the computational time required by the available R implementations of joint models, our study showed that the use of an approximate method at a genome-wide scale, such as TS, might represent a good compromise between accuracy and computational time. 
TS could be used to overcome the computational burden of current joint likelihood methods by exploiting available R packages performing the two steps, LME and CoxPH, and could help filter out SNPs with low or undetectable associations during a first preliminary scan. 
However, depending on the parameters of the dataset (sample size, incidence rate, number of measures), a joint likelihood method always has to be preferred over TS when one wants to obtain accurate estimation of parameters $\gamma$ and $\alpha$, which describe the SNP effect on the trajectory of FPG and on the time-to-onset of T2D, resp. 
Although we computed the theoretical statistical power to detect a joint genetic effect  $\beta\gamma+\alpha$ based [@chen_sample_2011], we did not test this effect at the genome-wide scale due to its computational burden. 
The joint SNP effect can be tested using a likelihood ratio test comparing the full joint model, i.e. with a SNP effect included in both sub-models, to the joint model without a SNP effect in the survival sub-model, as implemented in the package "JM" [@rizopoulos_jm:_2010; @R-JM].
Finally, we would like to reemphasize that using parallel and grid computing approaches will help reduce the global computational time when applied at a genome-wide scale (i.e. with millions of SNPs).

In our real data application, rs17747324 showed consistent results with the DIAGRAM consortium for both $\alpha$ and $\gamma$ (Table \@ref(tab:resultsConsistent)), but for rs10830963 showed an opposite effect on T2D risk compared to the effect reported in the MAGIC consortium for FPG ($\hat{\alpha}=`r results.consistent[grep("rs10830963", results.consistent[, "RSID"]), "alpha.estimate.DIAGRAM"]`$, $\textrm{p-value}=`r formatScientific(results.consistent[grep("rs10830963", results.consistent[, "RSID"]), "alpha.p.value.DIAGRAM"])`$). 
Results observed for _MTNR1B_ (rs10830963) in the French cohort D.E.S.I.R., albeit inconsistent with previous studies, may uncover some interesting peculiarities pertaining to T2D incident cases in this population. 
In the literature, SNPs in _MTNR1B_ were reported as being associated with higher FPG and T2D risk, but meta-analyses were performed on populations with different genetic backgrounds, and the two traits have never been jointly co-analysed. 
However, we realize that _MTNR1B_ associations identified in our study need to be confirmed and replicated in other cohorts, as they might be cohort-specific. 
Finally, a major limitation in our study comes from the low number of incident T2D cases in the D.E.S.I.R. cohort (`r format(N_IncT2D, big.mark = ",")` incident T2D cases in `r format(N_NoPrev, big.mark = ",")` individuals followed over 9 years).
