# List of tables {-}

`r if (knitr:::is_latex_output()) {'\\clearpage'} else {}`

(ref:SimulationSettings) `r paste0('Parameters and numerical values used for sensitivity analysis and simulations, based on results from ',  bestEffect[, "snp"], ' within gene _', bestEffect[, "closestgene"], '_.')`

```{r SimulationSettings, include = TRUE}
settings.table <- data.frame(matrix(NA, nrow = 9, ncol = 2, dimnames = list(NULL, c("Parameters", "Values"))))
settings.table[, "Parameters"] <- c("Number of participants ($n$)", "Number of measures ($m$)", "Diabetes incidence rate ($d$)", "Minor allele frequency ($f$)", "Random effects ($\\theta$)", "SNP effect on $Y_{ij}$ ($\\gamma$)", "SNP effect on $T_i$  ($\\alpha$)", "Association between $Y_{ij}$ and $T_i$ ($\\beta$)", "Error term ($\\epsilon$)")
settings.table[, "Values"] <- c(
    format(rawsettings$n+1, big.mark = ","),  
    4, 
    signif(rawsettings$d, digits = 3), 
    signif(rawsettings$maf, digits = 3),
    paste0("$\\sim\\mathcal{N}_2\\left (\\begin{bmatrix}", formatScientific(rawsettings$thetas[1]), "\\\\", formatScientific(rawsettings$thetas[2]), "\\end{bmatrix} , \\begin{bmatrix} ", formatScientific(rawsettings$Epsilon_thetas[1, 1]), " & ", formatScientific(rawsettings$Epsilon_thetas[1, 2]), " \\\\ ", formatScientific(rawsettings$Epsilon_thetas[2, 1]), " & ", formatScientific(rawsettings$Epsilon_thetas[2, 2]), " \\end{bmatrix} \\right )$"),
    signif(rawsettings$gamma, digits = 3), 
    signif(rawsettings$alpha[2], digits = 3), 
    signif(rawsettings$beta, digits = 3), 
    paste0("$\\sim\\mathcal{N}(0,", signif(rawsettings$sigma, digits = 3), "^2)$")
)
knitr::kable(
    settings.table, booktabs = TRUE,
    caption = '(ref:SimulationSettings)',
    format = "pandoc",
    align = c("l", "c")
)
```


(ref:sprof) Approximate computational times using function _system.time_ of R software. System time was computed ten times per sample size (number of individuals). Extrapolation is displayed for 100,000 SNPs

```{r sprof, include = TRUE}
sprof[, "mean.jm"] <- paste0(sprof[, "mean.jm"], "(", sprof[, "sd.jm"], ")")
sprof[, "mean.ts"] <- paste0(sprof[, "mean.ts"], "(", sprof[, "sd.ts"], ")")
sprof[, c("sd.jm", "sd.ts")]  <- NULL
sprof <- sprof[, c("n", "mean.jm", "mean.metabochip.jm", "mean.ts", "mean.metabochip.ts")]
sprof[, "mean.jm"] <- gsub("sec", "secs ", sprof[, "mean.jm"])
sprof[, "mean.jm"] <- gsub("secs", "", sprof[, "mean.jm"])
sprof[, "mean.ts"] <- gsub("sec", "secs ", sprof[, "mean.ts"])
sprof[, "mean.ts"] <- gsub("secs", "", sprof[, "mean.ts"])
sprof[, "mean.metabochip.jm"] <- gsub("days", "", sprof[, "mean.metabochip.jm"])
sprof[, "mean.metabochip.ts"] <- gsub("days", "", sprof[, "mean.metabochip.ts"])
sprof.out <- knitr::kable(
    rbind(c("Sample Size", "mean (sd) per SNP in seconds", "100K SNPs in days", "mean (sd) per SNP in seconds", "100K SNPs in days"), sprof), 
    booktabs = TRUE,
    col.names = c("", "Joint Model", "", "Two-Step Model", ""), 
    caption = '(ref:sprof)',
    format = "pandoc",
    align = rep("r", ncol(sprof))
)
sprof.out
```


(ref:PowerType1TCF7L2) `r paste0('RMSE, Type 1 Error and Power from default simulation settings for ',  bestEffect[, "snp"], ' (_', bestEffect[, "closestgene"], '_).')`

```{r PowerType1TCF7L2, include = TRUE}
TCF7L2_PowerType1[4, 1] <- "Î³"
TCF7L2_PowerType1[-1, 2] <- sprintf("%.3f", as.numeric(TCF7L2_PowerType1[-1, 2]), digits = 3)
TCF7L2_PowerType1[-1, 3] <- sprintf("%.3f", as.numeric(TCF7L2_PowerType1[-1, 3]), digits = 3)
TCF7L2_PowerType1[-1, 5] <- sprintf("%.3f", as.numeric(TCF7L2_PowerType1[-1, 5]), digits = 3)
TCF7L2_PowerType1[-1, 6] <- sprintf("%.3f", as.numeric(TCF7L2_PowerType1[-1, 6]), digits = 3)
TCF7L2_PowerType1[1, c(3, 4, 6, 7)] <- c("Type 1 Error", "Power", "Type 1 Error", "Power")
knitr::kable(
    TCF7L2_PowerType1[-3, ], 
    booktabs = TRUE,
    caption = '(ref:PowerType1TCF7L2)',
    format = "pandoc",
    align = rep("r", ncol(TCF7L2_PowerType1)),
    row.names = FALSE
)
```


<!-- (ref:resultstopsignunique) (\#tab:resultstopsignunique) List of loci found to be associated within the joint modelling framework with both FPG  and T2D risk; these loci have , previously  been shown as to be associated with FPG and/or T2D in the NHGRI GWAS Catalogue [@welter_nhgri_2014]. -->

```{r resultstopsignunique, include = TRUE}
resultstopsignunique <- results.topsign.unique
resultstopsignunique[, "alpha"] <- paste0(sprintf("%.3f", resultstopsignunique[, "alpha.estimate.JM"], digits = 3) , " ($",  formatScientific(resultstopsignunique[, "alpha.p.value.JM"]), "$)")
resultstopsignunique[, "gamma"] <- paste0(sprintf("%.3f", resultstopsignunique[, "gamma.estimate.JM"], digits = 4) , " ($",  formatScientific(resultstopsignunique[, "gamma.p.value.JM"]), "$)")
resultstopsignunique[, "beta"] <- paste0(sprintf("%.3f", resultstopsignunique[, "beta.estimate.JM"], digits = 2) , " ($",  formatScientific(resultstopsignunique[, "beta.p.value.JM"]), "$)")
resultstopsignunique <- resultstopsignunique[, c("RSID", "alpha", "gamma", "beta", "TheoPower")]
resultstopsignunique[, "term"] <- gsub("_. \\(.*", "", resultstopsignunique[, "RSID"])
resultstopsignunique <- merge(resultstopsignunique, results.sign[, c("RSID", "RiskAlleleFrequency")], by.x = "term", by.y = "RSID")[, -1]
colnames(resultstopsignunique) <- c("SNP (gene)", "$\\alpha$ (p-value)", "$\\gamma$ (p-value)", "$\\beta$ (p-value)", "Power ($\\beta\\gamma+\\alpha$)", "Risk Allele Frequency")
resultstopsignunique[, "Risk Allele Frequency"] <- round(resultstopsignunique[, "Risk Allele Frequency"], digits = 2)
resultstopsignunique <- resultstopsignunique[order(resultstopsignunique[, "Power ($\\beta\\gamma+\\alpha$)"], decreasing = TRUE), ]

rownames(resultstopsignunique) <- NULL
out <- capture.output(pander::pandoc.table(
    t = resultstopsignunique,
    caption = '(\\#tab:resultstopsignunique) List of loci found to be associated within the joint modelling framework with both FPG and T2D risk, previously shown as associated with FPG and/or T2D risk in the NHGRI GWAS Catalogue [@welter_nhgri_2014].',
    style = "simple",
    keep.line.breaks = TRUE,
    emphasize.rownames = FALSE,
    split.tables = Inf
))
asis_output(paste(c(out[length(out)-1], out[-((length(out)-2):length(out))]), collapse = "\n"))
```


<!-- (ref:resultsConsistent) (\#tab:resultsConsistent) `r paste0('Effect sizes on FPG and T2D risk estimated using joint modelling (JM) and TS (Two-Step). Comparison is shown with effect sizes as reported by consortia meta-analyses in genes ', gsub(".*\\((.*)\\)", "\\1", results.consistent[1, "RSID"]), ' and ', gsub(".*\\((.*)\\)", "\\1", results.consistent[1, "RSID"]), '. ')` -->

```{r resultsConsistent, include = TRUE}
result.consis <- results.consistent
result.consis[, "alpha.JM"] <- paste0(result.consis[, "alpha.estimate.JM"], " ($", formatScientific(result.consis[, "alpha.p.value.JM"]), "$)")
result.consis[, "gamma.JM"] <- paste0(result.consis[, "gamma.estimate.JM"], " ($", formatScientific(result.consis[, "gamma.p.value.JM"]), "$)")
result.consis[, "beta.JM"] <- paste0(result.consis[, "beta.estimate.JM"], " ($", formatScientific(result.consis[, "beta.p.value.JM"]), "$)")
result.consis[, "gamma.MAGIC"] <- paste0(result.consis[, "gamma.estimate.MAGIC"], " ($", formatScientific(result.consis[, "gamma.p.value.MAGIC"]), "$)")
result.consis[, "alpha.DIAGRAM"] <- paste0(result.consis[, "alpha.estimate.DIAGRAM"], " ($", formatScientific(result.consis[, "alpha.p.value.DIAGRAM"]), "$)")
result.consis <- result.consis[, c("RSID", "alpha.JM", "alpha.DIAGRAM", "gamma.JM", "gamma.MAGIC", "beta.JM")]
result.consis[, "RSID"] <- gsub("_", "\\_", result.consis[, "RSID"])

result.consis <- rbind(
    c("SNP (gene)", rep(c("$\\alpha$ (p-value)", "$\\gamma$ (p-value)", "$\\beta$ (p-value)"), each = 2)),
    c("", "JM (D.E.S.I.R.)", "DIAGRAM", "JM (D.E.S.I.R.)", "MAGIC", "JM (D.E.S.I.R.)"),
    result.consis
)

colnames(result.consis) <- result.consis[1, ]
resultconsis <- result.consis[-1, ]
rownames(resultconsis) <- NULL
out <- capture.output(pander::pandoc.table(
    t = resultconsis,
    caption = paste0('(\\#tab:resultsConsistent) Effect sizes on FPG and T2D risk estimated using JM and TS (Two-Step). Comparison is shown with effect sizes as reported by consortia meta-analyses in genes ', gsub(".*\\((.*)\\)", "\\1", results.consistent[1, "RSID"]), ' and ', gsub(".*\\((.*)\\)", "\\1", results.consistent[2, "RSID"]), '. '), # '(ref:resultsConsistent)',
    style = "simple",
    emphasize.rownames = FALSE,
    split.tables = Inf
))
out <- c(out[length(out)-1], out[-c(1, (length(out)-2):length(out))])
asis_output(paste(out, collapse = "\n"))
```

`r if (knitr:::is_latex_output()) {'\\clearpage'} else {}`
